app/models/conversion_host.rb

```diff
@@ -53,7 +54,7 @@ def source_transport_method
 
   def ipaddress(family = 'ipv4')
     return address if address.present? && IPAddr.new(address).send("#{family}?")
-    resource.ipaddresses.detect { |ip| IPAddr.new(ip).send("#{family}?") }
+    resource&.ipaddresses&.detect { |ip| IPAddr.new(ip).send("#{family}?") }
```

Comment:
Wouldn't it be possible to leverage the `ipaddress` attribute of the `Host` class, as it does some checks.

Reply:
The Host class has both an `ipaddress` and `ipaddresses` method as far as I can gather. Do they not do the same checks?

Reply:
`ipaddresses` returns all the IP addresses of the host, while `ipaddress` returns the IP address used for discovery. So, I have the feeling that `ipaddress` is more reliable. @agrare do you confirm ?

Reply:
The ipaddresses method goes through `host.hardware`. We can change it, but it will mean updating the specs elsewhere, too, as that will be a behavior change.

Reply:
As per a gitter conversation, we decided to leave this alone for now.